bulk_indexer.go

```diff
@@ -348,7 +351,10 @@ func (b *bulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 					// loop until we've seen the start newline
 					for seen+newlines < startln {
 						seen += newlines
-						n, _ := gr.Read(buf[:cap(buf)])
+						n, err := gr.Read(buf[:cap(buf)])
+						if err != nil && err != io.EOF {
```

Comment:
You only return early if err != nil and err != EOF

If err == EOF we haven't yet seen the start lewline, this will never exit. This shouldn't happen if the contents of `copyBuf` are what we expect, so it depends how defensive you want to be.